src/Game/DrawCardButton.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useMemo, useState } from "react";
+import { supabase } from "../supabase_client";
+
+const DrawCardButton = (props) => {
+  const {
+    currentPlayerLocationID,
+    myPlayerData,
+    currentTurnPlayer,
+    playerName,
+    hasPlayerMovedLocations,
+    setHasPlayerDrawnCard,
+    hasPlayerDrawnCard,
+  } = props;
+  const [availableCardsList, setAvailableCardsList] = useState([]);
+
+  const fetchLocationSpecificCards = async () => {
+    const { data } = await supabase
+      .from("CardList")
+      .select("cardID, name, description")
+      .eq("locationID", currentPlayerLocationID);
+    setAvailableCardsList(data);
+  };
+
+  useEffect(() => {
+    fetchLocationSpecificCards();
+  }, [currentPlayerLocationID]);
+
+  const randomCard = useMemo(() => {
+    return availableCardsList[
+      Math.floor(Math.random() * availableCardsList.length)
+    ];
+  }, [availableCardsList]);
```

Comment:
If you aren't displaying something from randomCard and you don't need it persisted across renders then you don't needa put it in a useMemo.

Seems like you can just move this into `sendCardToInventory`